tools/c7n_azure/c7n_azure/resources/cost_management_export.py

```diff
@@ -0,0 +1,163 @@
+        enum_spec = ('exports', 'list', None)
+        default_report_fields = (
+            'name',
+            'location',
+            'resourceGroup',
+        )
+        resource_type = 'Microsoft.Compute/images'
+
+        @classmethod
+        def extra_args(cls, resource_manager):
+            scope = '/subscriptions/{0}'\
+                .format(resource_manager.get_session().get_subscription_id())
+            return {'scope': scope}
+
+
+@CostManagementExport.filter_registry.register('last-execution')
+class CostManagementExportFilterLastExecution(Filter):
+    """ Find Cost Management Exports with last execution more than X days ago.
+
+    :example:
+
+    Returns all cost exports that didn't run in last 30 days.
+
+    .. code-block:: yaml
+
+        policies:
+          - name: find-stale-management-exports
+            resource: azure.cost-management-export
+            filters:
+              - type: last-execution
+                age: 30
+    """
+
+    schema = type_schema(
+        'last-execution',
+        required=['age'],
+        **{
+            'age': {'type': 'integer'}
+        }
+    )
+
+    def process(self, resources, event=None):
+        self.client = self.manager.get_client()
+        self.scope = 'subscriptions/{0}'.format(self.manager.get_session().get_subscription_id())
+        self.min_date = datetime.datetime.now() - datetime.timedelta(days=self.data['age'])
+
+        result, _ = ThreadHelper.execute_in_parallel(
+            resources=resources,
+            event=event,
+            execution_method=self._check_resources,
+            executor_factory=self.executor_factory,
+            log=log
+        )
+
+        return result
+
+    def _check_resources(self, resources, event):
+        result = []
+
+        for r in resources:
+            if gap('last-execution') in r:
+                continue
+            history = self.client.exports.get_execution_history(self.scope, r['name'])
+
+            # Include exports that has no execution history
+            if len(history.value) == 0:
+                r[gap('last-execution')] = 'None'
+                result.append(r)
+                continue
+
+            last_execution = max(history.value, key=lambda a: a.submitted_time)
+            if last_execution.submitted_time.date() <= self.min_date.date():
+                r[gap('last-execution')] = last_execution.serialize(True)
+                result.append(r)
+
+        return result
+
+
+@CostManagementExport.action_registry.register('execute')
+class CostManagementExportActionExecute(AzureBaseAction):
+    """ Trigger Cost Management Export execution
+
+    Known issues:
+
+    If you see an error
+    ``Error: (400) A valid email claim is required. Email claim is missing in the request header.``
+    please ensure used Service Principal has proper email configured.
+
+    :example:
+
+    Find all exports with last execution more than 30 days and trigger manual execution.

```

Comment:
(minor) 

Find all exports that have not been executed in the last 30 days and then trigger a manual export.
or
Find stale exports that are older than 30 days and trigger a manual export.